### PR TITLE
Feat/last three game stat methods

### PR DESCRIPTION
**Pull request by RodrigoACG**

### CODE REVIEW

File: lib/stat_generator.rb [hunk 1]
```diff
@@ -48,19 +48,74 @@ def total_ties
       game.away_goals == game.home_goals
     end
   end
-
+  
   def percentage_ties
     (total_ties / count_of_games.to_f).round(2)
+  end 
+  
+  def season_names
```
Comment (JaredMHobson) [inline lib/stat_generator.rb:56]: This accomplishes the same as the `seasons` method. I prefer this one but okay with keeping either. One of them will need to go before merging though.

File: lib/stat_generator.rb [hunk 2]
```diff
@@ -48,19 +48,74 @@ def total_ties
...
 
+  def average_goals_per_season(season)
+    #total num of goals by that season devided by total number of games that season had
+    (total_goals_by_season(season)) / count_of_games_by_season[season]
+  end
+  
+  #I need to get each game from each season and see the average from each season. 
+  # first i need to go through each game and sort them by season. @games.each do |game|  game.count_of_games_by_season 
+  #then nested??? something like (@games.each do |game| game.average_goals_per_season ) or something like total_goals_per_season / count_of_games_by_season
+  #then i need to see the averge score of each game.
```
Comment (JaredMHobson) [inline lib/stat_generator.rb:96]: Need to remove the comments as we clean things up

File: lib/stat_generator.rb [hunk 3]
```diff
@@ -48,19 +48,74 @@ def total_ties
...
+  def average_goals_by_season
+    goals_by_season = Hash.new(0)
+    # @games.each do |game|
+    #   season_names.each do |season|
+    #     require 'pry'; binding.pry
+    #     goals_by_season[season] = average_goals_per_season(season) if game.season == season
+    #   end
+    # end
+     season_names.each do |season|
+      goals_by_season[season] = average_goals_per_season(season).round(2)
```
Comment (JaredMHobson) [inline lib/stat_generator.rb:106]: Nicely done using the existing methods

File: lib/stat_generator.rb [hunk 4]
```diff
@@ -48,19 +48,74 @@ def total_ties
...
   def percentage_ties
     (total_ties / count_of_games.to_f).round(2)
+  end 
+  
+  def seasons
+    @games.map do |game|
+      game.season
+    end.uniq
+  end
+
```
Comment (NeilTheSeal) [inline lib/stat_generator.rb:61]: I like this seasons method. I think it is more efficient than the version I wrote

**Pull request merged**